pre-commit.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
```

Comment:
what do we do with windows users?
docker and docker compose commands also work on a windows machine.
But a shell script won't run.

Maybe the simplest solution is already in the english translation with the docker compose run --rm commands chained together

Reply:
> what do we do with windows users?

Honestly, my approach was to ignore that. There are no setup instructions for Windows (except for wsl and this works on wsl). I don't think anyone set up the ecamp environment on pure Windows ever; since I would expect there to be some Windows-specific discussions (like there are with Mac).

In the chapter [Code formatting](https://github.com/DeNic0la/ecamp3/blob/contribution-documentation/CONTRIBUTING.md#code-formatting-art) there is a very clear recommendation to use IDEA-based code formatting. As an **alternative** there are the 'docker compose run/execute'-commands. As a second alternative, there is a **suggestion** to add a git hook. As an **example** for that githook there is a reference to that shell script. 

I would argue an **example** githook is still an improvement even tho there is no pure Windows version (like everywhere else in the project). There is a recommended way and an alternative (both working on windows AFAIK). Developing on Windows is a wyld choice and if someone chooses to do so they have to modify the example (which can probably be done by ChatGPT) to meet their specific needs. This is completely fine. I am confident finding a way to format your code will be the least of your worries when developing ecamp3 on pure Windows.